beetsplug/lyrics.py

```diff
@@ -659,26 +671,133 @@ def commands(self):
             help=u'print lyrics to console',
         )
         cmd.parser.add_option(
+            u'-r', u'--write-rst', dest='writerst',
+            action='store', default='.', metavar='dir',
+            help=u'write lyrics to given directory as RST files',
+        )
+        cmd.parser.add_option(
             u'-f', u'--force', dest='force_refetch',
             action='store_true', default=False,
             help=u'always re-download lyrics',
         )
+        cmd.parser.add_option(
+            u'-l', u'--local', dest='local_only',
+            action='store_true', default=False,
+            help=u'do not fetch missing lyrics',
+        )
 
         def func(lib, opts, args):
             # The "write to files" option corresponds to the
             # import_write config value.
             write = ui.should_write()
+            if opts.writerst:
+                self.writerst_indexes(opts.writerst)
             for item in lib.items(ui.decargs(args)):
-                self.fetch_item_lyrics(
-                    lib, item, write,
-                    opts.force_refetch or self.config['force'],
-                )
-                if opts.printlyr and item.lyrics:
-                    ui.print_(item.lyrics)
-
+                if not opts.local_only and not self.config['local']:
+                    self.fetch_item_lyrics(
+                        lib, item, write,
+                        opts.force_refetch or self.config['force'],
+                    )
+                if item.lyrics:
+                    if opts.printlyr:
+                        ui.print_(item.lyrics)
+                    if opts.writerst:
+                        self.writerst(opts.writerst, item)
+            if opts.writerst:
+                # flush last artist
+                self.writerst(opts.writerst, None)
+                ui.print_(u'RST files generated. to build, use one of:')
+                ui.print_(u'  sphinx-build -b html %s _build/html'
+                          % opts.writerst)
+                ui.print_(u'  sphinx-build -b epub %s _build/epub'
+                          % opts.writerst)
+                ui.print_((u'  sphinx-build -b latex %s _build/latex '
+                           u'&& make -C _build/latex all-pdf')
+                          % opts.writerst)
         cmd.func = func
         return [cmd]
 
+    def writerst(self, directory, item):
+        """Write the item to an RST file
+
+        this will keep state (in the `rst` variable) in order to avoid
+        writing continuously to the same files
```

Comment:
fine with this too, feel free to correct, otherwise i may take a day or two to catchup here too.